src/Symfony/Contracts/HttpClient/ResponseInterface.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Contracts\HttpClient;
+
+use Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\ExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\RedirectionExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+
+/**
+ * A (lazily retrieved) HTTP response.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @throws RedirectionExceptionInterface On a 3xx when getStatusCode() has not been called
+ * @throws ClientExceptionInterface      On a 4xx when getStatusCode() has not been called
+ * @throws ServerExceptionInterface      On a 5xx when getStatusCode() has not been called
+ * @throws TransportExceptionInterface   when a network error occurs
+ */
+interface ResponseInterface
+{
+    /**
+     * Gets the HTTP status code of the response.
+     */
+    public function getStatusCode(): int;
+
+    /**
+     * @return string[][] The headers of the response keyed by header names in lowercase
+     */
+    public function getHeaders(): array;
+
+    /**
+     * Gets the response body as a string.
+     */
+    public function getContent(): string;
+
+    /**
+     * Returns info coming from the transport layer.
+     *
+     * This method SHOULD NOT throw any ExceptionInterface and SHOULD be non-blocking.
+     *
+     * The following info MUST be returned:
+     *  - raw_headers - an array modelled after the special $http_response_header variable
+     *  - redirect_count - the number of redirects followed while executing the request
+     *  - redirect_url - the resolved location of redirect responses, null otherwise
+     *  - start_time - the time when the request was sent
+     *  - http_code - the last response code or 0 when it is not known yet
+     *  - error - the error message when the transfer was aborted, null otherwise
+     *  - data - the value of the "data" request option, null if not set
+     *  - url - the last effective URL of the request
+     *
+     * When the "capture_peer_cert_chain" option is true, the "peer_certificate_chain"
+     * attribute SHOULD list the peer certificates as an array of OpenSSL X.509 resources.
+     *
+     * Other info SHOULD be named after curl_getinfo()'s associative return value.
+     */
+    public function getInfo(string $type = null);
```

Comment:
missing return type, either in the signature or the phpdoc.

Reply:
added in #30413